File: c/leetcode/动态规划/53 最大子序和/official.c
int maxSubArray(int* nums, int numsSize){
    int m[numsSize];
    m[0] = nums[0];
    for(int i = 1; i < numsSize; i++){
        m[i] = m[i-1] + nums[i];
        if(m[i] < nums[i])
            m[i] = nums[i];
    }
    int max = INT_MIN;
    for(int i = 0; i < numsSize; i++)
        max = max < m[i] ? m[i] : max;
    return max;
}

//关键始终是在写出递推公式上，如何分析最优子结构是个关键呀！

File: leetcode/动态规划/55 跳跃游戏/official.go
func canJump(nums []int) bool {
	len := len(nums)
	rightmost := 0
	for i := 0; i < len; i++ {
		if i <= rightmost {
			rightmost = max(i+nums[i], rightmost)
		} else {
			return false
		}
		if rightmost >= len {
			return true
		}

	}
	return true
}

func max(a int, b int) int {
	if a > b {
		return a
	}
	return b
}

//官方思路，贪心算法，感觉思路都比较类似，就是捋没捋清楚的问题。没弄清楚就是我那个倒着来恶心人的题解，弄清楚了就是这个很顺畅的题解。